src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexBetween(index);
+        if (index == size) {
+            add(value);
+        } else {
+            link(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+
```

Comment:
```suggestion
```
if list is null, should throw NPE
![image](https://github.com/mate-academy/jv-linked-list/assets/49767049/b4ae45ee-f4c6-44dc-b461-8aaa5744941e)